platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateRoot.java

```diff
@@ -1082,4 +1101,51 @@ public void computeHash() {
             Thread.currentThread().interrupt();
         }
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void createSnapshot(Path targetPath) {
+        Time time = platformContext.getTime();
+        final long start = time.nanoTime();
+        throwIfMutable();
+        throwIfDestroyed();
+        createSnapshot(this, targetPath);
+        snapshotMetrics.getWriteStateToDiskTimeMetric().update(TimeUnit.NANOSECONDS.toMillis(time.nanoTime() - start));
+    }
+
+    static void createSnapshot(MerkleRoot merkleRoot, Path targetPath) {
```

Comment:
This file is huge. Might be a good idea to split this (and the other) static helper methods into a separate utility class

Reply:
Done, The logic is in `MerkleTreeSnapshotWriter` now